p2p/raintree/router.go

```diff
@@ -0,0 +1,61 @@
+package raintree
+
+import (
+	"log"
+	"strings"
+)
+
+type router struct {
+	network *rainTreeNetwork
+}
+
+func newRouter(n *rainTreeNetwork) *router {
```

Comment:
TOL (Thinking Out Loud): I like having the `router.go` file and how the logic is encapsulated, but given this 1:1 mapping between `rainTreeNetwork` and `router`, why not just make the receiver of all these functions `rainTreeNetwork` (since it's in the raintree package anyhow) and avoid the extra structure and creator?